src/app/pages/products/components/filters/filter-panel/filter-panel.component.ts

```diff
@@ -32,7 +32,13 @@ export class FilterPanelComponent implements OnInit {
     private cdr: ChangeDetectorRef,
   ) {}
 
-  ngOnInit() {
+  async ngOnInit(): Promise<void> {
+    this.filterDefinitions = await firstValueFrom(this.filterFacadeService.getFilterDefinitions());
+
+    this.filterDefinitions.forEach((definition: IFilterDefinition) => {
+      this.form.addControl(definition.id, new FormControl([]));
+    });
```

Comment:
Move to some method here for not stopping the next code by removing async operation from ngOnInit